In [1]:
import numpy as np

# General understanding

### Following my example, convert the energy values for the following food items from kcal g⁻¹ to the SI equivalent in MJ kg⁻¹. *Be sure to round to the correct number of significant figures.* **(1 mark per answer, 3 total)**

<table style='float:left'>
    <tr>
        <th style='text-align:left'>Food</th><th style='text-align:center'>Energy<br>kcal (100 g)⁻¹</th>
    </tr>
    <tr>
        <td style='text-align:left'>Tic Tacs</td><td>386</td>
    </tr>
    <tr>
        <td style='text-align:left'>Bitesize shredded wheat</td><td>149</td>
    </tr>
    <tr>
        <td style='text-align:left'>Oatmeal</td><td>370</td>
    </tr>
    <tr>
        <td style='text-align:left'>Vegetable oil</td><td>900</td>
    </tr>
</table>

In [2]:
print('Tic Tacs: {:.1f} MJ kg⁻¹'.format(386*4.184/100))

Tic Tacs: 16.2 MJ kg⁻¹


In [3]:
print('Bitesize shredded wheat: {:.2f} MJ kg⁻¹'.format(149*4.184/100))
print('Oatmeal: {:.1f} MJ kg⁻¹'.format(370*4.184/100,1))
print('Vegetable oil: {:.1f} MJ kg⁻¹'.format(900*4.184/100,1))

Bitesize shredded wheat: 6.23 MJ kg⁻¹
Oatmeal: 15.5 MJ kg⁻¹
Vegetable oil: 37.7 MJ kg⁻¹


# Coffee cup calorimetry

## Specific heat capacity determination

### Use your Jupyter notebook to calculate the missing values in red. Refer to Blackboard to get the values you should use in the spots marked with blue capital letters.

*I've filled in model values that are different from any of those that are assigned.*

<table style='float:left'>
    <tr>
        <th style='text-align:left'>measurement</th><th>value</th><th style='text-align:center'>absolute<br>uncertainty</th>
    </tr>
    <tr>
        <td style='text-align:left'>mass of empty coffee cups (g)</td><td style="color:blue">M_CUP_A</td><td>±0.001</td>
    </tr>
    <tr>
        <td style='text-align:left'>mass of water-filled cups (g)</td><td style="color:blue">M_CUP_WATER_A</td><td>±0.001</td>
    </tr>
    <tr>
        <td style='text-align:left'>mass of cups with water and sample (g)</td><td style="color:blue">M_CUP_WATER_SAMPLE_A</td><td>±0.01</td>
    </tr>
    <tr>
        <td style='text-align:left'>initial temperature water (°C)</td><td style="color:blue">T_WATER_A</td><td>±0.3</td>
    </tr>
    <tr>
        <td style='text-align:left'>initial temperature sample (°C)</td><td style="color:blue">T_SAMPLE_A</td><td>±2</td>
    </tr>
    <tr>
        <td style='text-align:left'>final temperature (°C)</td><td style="color:blue">T_FINAL_A</td><td>±0.3</td>
    </tr>
    <tr>
        <td style='text-align:left'>mass water (g)</td><td style="color:red">m_water</td><td style="color:red">err_m_water</td>
    </tr>
    <tr>
        <td style='text-align:left'>mass sample (g)</td><td style="color:red">m_sample</td><td style="color:red">err_m_sample</td>
    </tr>
    <tr>
        <td style='text-align:left'>temperature change water (K)</td><td style="color:red">delta_T_water</td><td style="color:red">err_delta_T_water</td>
    </tr>
    <tr>
        <td style='text-align:left'>temperature change sample (K)</td><td style="color:red">delta_T_sample</td><td style="color:red">err_delta_T_sample</td>
    </tr>
    <tr>
        <td style='text-align:left'>heat capacity of liquid water (kJ kg⁻¹ K⁻¹)</td><td>4.184</td><td>0.004</td>
    </tr>
    <tr>
        <td style='text-align:left'>heat capacity of sample (kJ kg⁻¹ K⁻¹)</td><td style="color:red">cp_sample</td><td style="color:red">err_cp_sample</td>
    </tr>
</table>

### Calculate the value of *m_water*, *m_sample*, *delta_T_water*, and *delta_T_sample* **(1 mark each, 4 total)**

In [4]:
M_CUP_A = 5.207 # mass empty coffee cups
M_CUP_WATER_A = 64.823 # same cups with water
M_CUP_WATER_SAMPLE_A = 228.100 # same cups with water and sample
T_WATER_A = 15.2 # temperature of tap water at the start
T_SAMPLE_A = 99 # temperature of the sample taken out of the oven
T_FINAL_A = 22.4 # final temperature of the whole lot
cp_water = 4.184 # heat capacity of the water from the table

# named new variable based on what's in the table
err_M_CUP_A = 0.001 # uncertainty in the mass of the coffee cups
err_M_CUP_WATER_A = 0.001 # uncertainty in the mass of the coffee cups + water
err_M_CUP_WATER_SAMPLE_A = 0.01 # uncertainty in the mass of the coffee cups + water + sample
err_T_WATER_A = 0.3 # uncertainty in the initial water temperature
err_T_SAMPLE_A = 2. # uncertainty in the sample temperature
err_T_FINAL_A = 0.3 # uncertainty in the final temperature of the everything together
err_cp_water = 0.004 # uncertainty in the heat capacity of the water (table)

# the calculations
m_water = M_CUP_WATER_A - M_CUP_A # mass of the water, by difference
m_sample = M_CUP_WATER_SAMPLE_A - M_CUP_WATER_A # mass of the sample, by difference
delta_T_water = T_FINAL_A - T_WATER_A # temperature change of the water
delta_T_sample = T_FINAL_A - T_SAMPLE_A # temperature change of the sample

# output the results
print('Mass of water: {:.3f} g'.format(m_water))
print('Mass of sample: {:.3f} g'.format(m_sample))
print('∆T water: {:.1f} °C'.format(delta_T_water))
print('∆T sample: {:.0f} °C'.format(delta_T_sample))

Mass of water: 59.616 g
Mass of sample: 163.277 g
∆T water: 7.2 °C
∆T sample: -77 °C


### Calculate the associated uncertainty values for *m_water*, *m_sample*, *delta_T_water*, and *delta_T_sample* (that is, *err_m_water*, *err_m_sample*, *err_delta_T_water* and *err_delta_T_sample*) **(2 marks each, 8 total)**

In [5]:
# all of these are done using absolute errors because the operation is addition/subtraction
err_m_water = np.sqrt(err_M_CUP_WATER_A**2 + err_M_CUP_A**2) # uncertainty in the mass of the water
err_m_sample = np.sqrt(err_M_CUP_WATER_SAMPLE_A**2 + err_M_CUP_WATER_A**2) # uncertainty in the mass of the sample
err_delta_T_water = np.sqrt(err_T_FINAL_A**2 + err_T_WATER_A**2) # uncert in the temperature change of the water
err_delta_T_sample = np.sqrt(err_T_FINAL_A**2 + err_T_SAMPLE_A**2) # uncert in the temperature change of the sample

# output the results
print('Uncertainty in the mass of water: {:f} g'.format(err_m_water))
print('Uncertainty in the mass of the sample: {:f} g'.format(err_m_sample))
print('Uncertainty in the temperature change of the water: {:f} °C'.format(err_delta_T_water))
print('Uncertainty in the temperature change of the sample: {:f} °C'.format(err_delta_T_sample))

Uncertainty in the mass of water: 0.001414 g
Uncertainty in the mass of the sample: 0.010050 g
Uncertainty in the temperature change of the water: 0.424264 °C
Uncertainty in the temperature change of the sample: 2.022375 °C


### Calculate the value of the heat capacity of the sample, *cp_sample* **(3 marks)**

In [6]:
denom_A = m_sample * delta_T_sample # the denominators shown in equations 7 and 10 have this common factor
cp_sample = -m_water * cp_water * delta_T_water / denom_A # heat capacity of the sample (equation 7)
print('Heat capacity of sample: {:f} kJ kg⁻¹ K⁻¹'.format(cp_sample)) # no rounding yet

Heat capacity of sample: 0.143593 kJ kg⁻¹ K⁻¹


### Calculate the uncertainty in the heat capacity (that is, *err_cp_sample*) **(8 marks)**

In [7]:
# working out each part under the square root separately to make it clearer
num_A = np.zeros(6)
num_A[0] = - err_m_water * cp_water * delta_T_water
num_A[1] = - err_cp_water * m_water * delta_T_water
num_A[2] = err_T_FINAL_A * cp_water * m_water * (T_SAMPLE_A - T_WATER_A) / delta_T_sample
num_A[3] = err_T_WATER_A * cp_water * m_water
num_A[4] = err_m_sample * cp_water * m_water * delta_T_water / m_sample
num_A[5] = - err_T_SAMPLE_A * cp_water * m_water * delta_T_water / delta_T_sample
print(num_A)
err_cp_sample = np.sqrt(np.sum(num_A ** 2)) / abs(denom_A)
print('Uncertainty in the heat capacity of the sample: {:f} kJ kg⁻¹ K⁻¹'.format(err_cp_sample))

[-4.26029007e-02 -1.71694080e+00 -8.18636327e+01  7.48300032e+01
  1.10540820e-01  4.68908636e+01]
Uncertainty in the heat capacity of the sample: 0.009629 kJ kg⁻¹ K⁻¹


### Replace the *cp_sample* and *err_cp_sample* to express the final heat capacity of the material with its associated precision. Marks are awarded for correct rounding. **(2 marks)**

(*0.1436* ± *0.0096*) kJ kg⁻¹ K⁻¹

### Use your calculation for heat capacity to identify what material you measured using https://www.engineeringtoolbox.com/specific-heat-capacity-d_391.html. Take into account the uncertainty you determined for your measurement when giving your explanation. **(1 mark)**

<div class="alert alert-block alert-info">
    <strong>Pleasse note</strong>
    <br>
    <ul>
        <li>The table below is what was used to generate the values in the problems, but the identity does not need to match.</li>
    </ul>
    <table>
        <tr>
            <th style='text-align:left'>material</th><th>specific heat capacity<br>(kJ kg⁻¹ K⁻¹)</th>
        </tr>
        <tr>
            <td style='text-align:left'>steel</td><td>0.49</td>
        </tr>
        <tr>
            <td style='text-align:left'>brass (60% Cu, 40% Zn)</td><td>0.38</td>
        </tr>
        <tr>
            <td style='text-align:left'>duralumin</td><td>0.92</td>
        </tr>
        <tr>
            <td style='text-align:left'>porcelain</td><td>1.08</td>
        </tr>
        <tr>
            <td style='text-align:left'>PET</td><td>1.25</td>
        </tr>
        <tr>
            <td style='text-align:left'>nylon-6,6</td><td>1.70</td>
        </tr>
    </table>
</div>

### Discuss one possible sources of systematic error. **(1 mark)**

<div class="alert alert-block alert-info">
    <ul>
        <li>good reasons for systematic error include:</li>
        <ul>
            <li>not adiabatic</li>
            <li>evaporation</li>
            <li>thermometer/balance calibration</li>
        </ul>
        <li>less good reasons for systematic error include:</li>
        <ul>
            <li>Joule heating from stirring: not likely to be significant</li>
            <li>oven temperature poorly known: not a dominant source of uncertainty</li>
            <li>parallax error on reading: it’s hard to correct poor practice</li>
        </ul>
    </ul>
</div>

### Suggest how the size of that systematic error could be tested. **(2 marks)**

<div class="alert alert-block alert-info">
    <strong>See below</strong><br>
        <ul>
            <li>not adiabatic: add more insulation, see how long it takes temperature to drop</li>
            <li>evaporation: put on a cover, check the difference</li>
            <li>thermometer/balance calibration: use another to check</li>
            <li>Joule heating from stirring: oven temperature poorly known: use integrated thermometer in sample, check the effect of uncertainty on uncertainty of final answer</li>
            <li>parallax error on reading: use digital thermometer</li>
        </ul>
    </ul>
</div>

## Heat of solution determination

### Give the adiabatic heat balance for the dissolution of salt in water. Assume that salt does not change the heat capacity of the water and that the salt and water start at the same temperature. The units for heat of solution are J (g salt)⁻¹. **(3 marks)**

$q = 0 = m_{\textrm{water}} C_{p}^{\textrm{water}} (T_{\textrm{f}} – T_{\textrm{i}}) + m_{\textrm{salt}} \Delta_{\textrm{soln}}H$

### Solve this equation for the heat of solution. **(4 marks)**

$\Delta_{\textrm{soln}}H = -\frac{m_{\textrm{water}} C_{p}^{\textrm{water}} (T_{\textrm{f}} – T_{\textrm{i}})}{m_{\textrm{salt}}}$

### Use your Jupyter notebook to calculate the missing values in red. Refer to Blackboard to get the values you should use in the spots marked with blue capital letters.

<table style='float:left'>
    <tr>
        <th style='text-align:left'>measurement</th><th>value</th><th style='text-align:center'>absolute<br>uncertainty</th>
    </tr>
    <tr>
        <td style='text-align:left'>mass of empty coffee cups (g)</td><td style="color:blue">M_CUP_B</td><td>±0.001</td>
    </tr>
    <tr>
        <td style='text-align:left'>mass of water-filled cups (g)</td><td style="color:blue">M_CUP_WATER_B</td><td>±0.001</td>
    </tr>
    <tr>
        <td style='text-align:left'>mass of cups with water and salt (g)</td><td style="color:blue">M_CUP_WATER_SALT_B</td><td>±0.001</td>
    </tr>
    <tr>
        <td style='text-align:left'>initial temperature water (°C)</td><td style="color:blue">T_WATER_I_B</td><td>±0.005</td>
    </tr>
    <tr>
        <td style='text-align:left'>final temperature (°C)</td><td style="color:blue">T_WATER_F_B</td><td>±0.005</td>
    </tr>
    <tr>
        <td style='text-align:left'>mass water (g)</td><td style="color:red">m_water</td><td style="color:red">err_m_water</td>
    </tr>
    <tr>
        <td style='text-align:left'>mass salt (g)</td><td style="color:red">m_salt</td><td style="color:red">err_m_salt</td>
    </tr>
    <tr>
        <td style='text-align:left'>temperature change water (K)</td><td style="color:red">delta_T</td><td style="color:red">err_delta_T</td>
    </tr>
    <tr>
        <td style='text-align:left'>heat capacity of liquid water (kJ kg⁻¹ K⁻¹)</td><td>4.184</td><td>0.004</td>
    </tr>
    <tr>
        <td style='text-align:left'>heat of solution (MJ kg⁻¹)</td><td style="color:red">Hsoln</td><td style="color:red">err_Hsoln</td>
    </tr>
</table>

### Calculate the values of *m_water*, *m_salt*, and *delta_T*. **(1 mark each, 3 marks total)**

In [8]:
# initialise values
M_CUP_B = 5.187
M_CUP_WATER_B = 107.367
M_CUP_WATER_SALT_B = 107.504
T_WATER_I_B = 18.599
T_WATER_F_B = 18.545

# calculate values by difference
m_water = M_CUP_WATER_B - M_CUP_B
m_salt = M_CUP_WATER_SALT_B - M_CUP_WATER_B
delta_T = T_WATER_F_B - T_WATER_I_B

# output values
print('Mass of water: {:.3f} g'.format(m_water))
print('Mass of salt: {:.3f} g'.format(m_salt))
print('Temperature change of solution: {:.3f} °C'.format(delta_T))

Mass of water: 102.180 g
Mass of salt: 0.137 g
Temperature change of solution: -0.054 °C


### Calculate the associated uncertainty values for these calculations, *err_m_water*, *err_m_salt*, and *err_delta_T*. **(2 marks each, 6 marks total)**

In [9]:
# initialise values
err_M_CUP_B = 0.001
err_M_CUP_WATER_B = 0.001
err_M_CUP_WATER_SALT_B = 0.001
err_T_WATER_I_B = 0.005
err_T_WATER_F_B = 0.005

# using absolute errors because values are determined by subtraction
err_m_water = np.sqrt(err_M_CUP_WATER_B ** 2 + err_M_CUP_B ** 2)
err_m_salt = np.sqrt(err_M_CUP_WATER_SALT_B ** 2 + err_M_CUP_WATER_B ** 2)
err_delta_T = np.sqrt(err_T_WATER_F_B ** 2 + err_T_WATER_I_B ** 2)

# output values
print('Uncertainty in mass of water: {:f} g'.format(err_m_water))
print('Uncertainty in mass of salt: {:f} g'.format(err_m_salt))
print('Uncertainty in temperature change of solution: {:f} °C'.format(err_delta_T))

Uncertainty in mass of water: 0.001414 g
Uncertainty in mass of salt: 0.001414 g
Uncertainty in temperature change of solution: 0.007071 °C


### Calculate the heat of solution for this salt, *Hsoln*. Note that the units need to be MJ kg⁻¹, not kJ kg⁻¹. (3 marks)

In [10]:
# calculate based on equation determined above
Hsoln = -cp_water * m_water * delta_T / m_salt / 1000 # divide by 1000 to get to MJ/kg

# output values
print('Heat of solution: {:f} MJ kg⁻¹'.format(Hsoln))

Heat of solution: 0.168512 MJ kg⁻¹


### Calculate the uncertainty in this heat of solution, *err_Hsoln*. **(9 marks)**

In [11]:
# working out each part under the square root separately to make it clearer
denom_B = -m_salt
num_B = np.zeros(5)
num_B[0] = err_m_water * cp_water * delta_T
num_B[1] = err_cp_water * m_water * delta_T
num_B[2] = err_T_WATER_F_B * cp_water * m_water
num_B[3] = -err_T_WATER_I_B * cp_water * m_water
num_B[4] = -err_m_salt * cp_water * m_water * delta_T / m_salt
print(num_B)
err_Hsoln = np.sqrt(np.sum(num_B ** 2)) / abs(denom_B) / 1000
print('Uncertainty in heat of solution: {:f} MJ kg⁻¹'.format(err_Hsoln))

[-3.19521755e-04 -2.20708800e-02  2.13760560e+00 -2.13760560e+00
  2.38311919e-01]
Uncertainty in heat of solution: 0.022135 MJ kg⁻¹


### Replace the *Hsoln* and *err_Hsoln* to express the final value for the heat of solution with its associated precision. Marks are awarded for correct rounding. **(2 marks)**

(*0.169* ± *0.022*) MJ (kg salt)⁻¹

## Latent heat capacity determination

### Solve the following heat balance equation for the heat of fusion of water. You'll find it easiest to copy the terms for the working equation. **(4 marks)**

$q = 0 = m_{\textrm{ice}} C_{p}^\textrm{ice} (T_{\textrm{fus}} - T_{\textrm{ice}}) + m_{\textrm{ice}} C_{p}^\textrm{water} (T_{\textrm{f}} – T_{\textrm{fus}}) + m_{\textrm{water}} C_{p}^\textrm{water} (T_{\textrm{f}} – T_{\textrm{water}}) + m_{\textrm{ice}} \Delta_{\textrm{fus}}H$

$\Delta_{\textrm{fus}}H = \frac{numerator}{denominator}$

<div class="alert alert-block alert-info">
    $\Delta_{\textrm{fus}}H = -\frac{m_{\textrm{ice}} C_{p}^\textrm{ice} (0\textrm{ °C} - T_{\textrm{ice}}) + m_{\textrm{ice}} C_{p}^\textrm{water} (T_{\textrm{f}} - 0\textrm{ °C}) + m_{\textrm{water}} C_{p}^\textrm{water} (T_{\textrm{water}} - 0\textrm{ °C})}{m_{\textrm{ice}}}$
    <br>
    <br>The temperatures are in °C and none of the temperature has an exponent other than one, so this simplifies to
    $\Delta_{\textrm{fus}}H = -\frac{m_{\textrm{ice}} (C_{p}^\textrm{water} T_{\textrm{f}} - C_{p}^\textrm{ice} T_{\textrm{ice}}) + m_{\textrm{water}} C_{p}^\textrm{water} (T_{\textrm{f}} - T_{\textrm{water}})}{m_{\textrm{ice}}} = \frac{m_{\textrm{ice}} C_{p}^\textrm{ice} T_{\textrm{ice}} - C_{p}^\textrm{water} (m_{\textrm{ice}} T_{\textrm{f}} + m_{\textrm{water}} (T_{\textrm{f}} - T_{\textrm{water}}))}{m_{\textrm{ice}}}$
    <br>
    <br>

### Use your Jupyter notebook to calculate the missing values in red. Refer to Blackboard to get the values you should use in the spots marked with blue capital letters.

<table style='float:left'>
    <tr>
        <th style='text-align:left'>measurement</th><th>value</th><th style='text-align:center'>absolute<br>uncertainty</th>
    </tr>
    <tr>
        <td style='text-align:left'>mass of empty coffee cups (g)</td><td style="color:blue">M_CUP_C</td><td>±0.001</td>
    </tr>
    <tr>
        <td style='text-align:left'>mass of water-filled cups (g)</td><td style="color:blue">M_CUP_WATER_C</td><td>±0.001</td>
    </tr>
    <tr>
        <td style='text-align:left'>mass of cups with water and melted ice (g)</td><td style="color:blue">M_CUP_WATER_ICE_C</td><td>±0.001</td>
    </tr>
    <tr>
        <td style='text-align:left'>initial temperature water (°C)</td><td style="color:blue">T_WATER_C</td><td>±0.3</td>
    </tr>
    <tr>
        <td style='text-align:left'>initial temperature ice (°C)</td><td style="color:blue">T_ICE_C</td><td>±2</td>
    </tr>
    <tr>
        <td style='text-align:left'>final temperature (°C)</td><td style="color:blue">T_FINAL_C</td><td>±0.3</td>
    </tr>
    <tr>
        <td style='text-align:left'>mass water (g)</td><td style="color:red">m_water</td><td style="color:red">err_m_water</td>
    </tr>
    <tr>
        <td style='text-align:left'>mass ice (g)</td><td style="color:red">m_ice</td><td style="color:red">err_m_ice</td>
    </tr>
    <tr>
        <td style='text-align:left'>temperature change ice to T<span style="font-size:60%;vertical-align:sub">fus</span> (K)</td><td style="color:red">delta_T1_ice</td><td style="color:red">errdelta_T1_ice</td>
    </tr>
    <tr>
        <td style='text-align:left'>temperature change T<span style="font-size:60%;vertical-align:sub">fus</span> to T<span style="font-size:60%;vertical-align:sub">final</span> (K)</td><td style="color:red">delta_T2_ice</td><td style="color:red">err_delta_T2_ice</td>
    </tr>
    <tr>
        <td style='text-align:left'>temperature change water to T<span style="font-size:60%;vertical-align:sub">final</span> (K)</td><td style="color:red">delta_T_water</td><td style="color:red">err_delta_T_water</td>
    </tr>
    <tr>
        <td style='text-align:left'>heat capacity of liquid water (kJ kg⁻¹ K⁻¹)</td><td>4.184</td><td>0.004</td>
    </tr>
    <tr>
        <td style='text-align:left'>heat capacity of solid water (kJ kg⁻¹ K⁻¹)</td><td>2.07</td><td>0.03</td>
    </tr>
    <tr>
        <td style='text-align:left'>latent heat of fusion of ice (kJ kg⁻¹)</td><td style="color:red">Hfus_water</td><td style="color:red">err_Hfus_water</td>
    </tr>
</table>

### Calculate the values of *m_water*, *m_ice*, *delta_T1_ice*, *delta_T2_ice* and *delta_T_water*. **(1 mark each, 5 marks total)**

In [12]:
# as before, initialise the values
M_CUP_C = 5.259
M_CUP_WATER_C = 60.853
M_CUP_WATER_ICE_C = 78.927
T_WATER_C = 54.9
T_ICE_C = -8.2
T_FINAL_C = 21.1
cp_ice = 2.07

# calculate all the required values by difference, keeping in mind we're assuming Tfus(H2O) is exactly 0 °C
m_water = M_CUP_WATER_C - M_CUP_C
m_ice = M_CUP_WATER_ICE_C - M_CUP_WATER_C
delta_T1_ice = 0 - T_ICE_C
delta_T2_ice = T_FINAL_C - 0
delta_T_water = T_FINAL_C - T_WATER_C

# output the values
print('Mass water: {:.3f} g'.format(m_water))
print('Mass ice: {:.3f} g'.format(m_ice))
print('Temperature change of ice to its melting point: {:.1f} °C'.format(delta_T1_ice))
print('Temperature change of melted ice to the final temperature: {:.1f} °C'.format(delta_T2_ice))
print('Temperature change of the water to its final temperature: {:.1f} °C'.format(delta_T_water))

Mass water: 55.594 g
Mass ice: 18.074 g
Temperature change of ice to its melting point: 8.2 °C
Temperature change of melted ice to the final temperature: 21.1 °C
Temperature change of the water to its final temperature: -33.8 °C


### Calculate the the associated uncertainty values: *err_m_water*, *err_m_ice*, *err_delta_T1_ice*, *err_delta_T2_ice* and *err_delta_T_water*. **(2 marks each, 10 marks total)**

In [13]:
# initialise values
err_M_CUP_C = 0.001
err_M_CUP_WATER_C = 0.001
err_M_CUP_WATER_ICE_C = 0.001
err_T_WATER_C = 0.3
err_T_ICE_C = 2.
err_T_FINAL_C = 0.3
err_cp_ice = 0.03

# all the values were differences, so work with absolute errors
err_m_water = np.sqrt(err_M_CUP_WATER_C ** 2 + err_M_CUP_C ** 2)
err_m_ice = np.sqrt(err_M_CUP_WATER_ICE_C ** 2 + err_M_CUP_WATER_C ** 2)
err_delta_T1_ice = np.sqrt(0 ** 2 + err_T_ICE_C ** 2)
err_delta_T2_ice = np.sqrt(err_T_FINAL_C ** 2 + 0 ** 2)
err_delta_T_water = np.sqrt(err_T_FINAL_C ** 2 + err_T_WATER_C ** 2)

# output the values
print('Uncertainty in the mass of the water: {:f} g'.format(err_m_water))
print('Uncertainty in the mass of the ice: {:f} g'.format(err_m_ice))
print('Uncertainty in the temperature change of ice to its melting point: {:f} °C'.format(err_delta_T1_ice))
print('Uncertainty in the temperature change of melted ice to the final temp.: {:f} °C'.format(err_delta_T2_ice))
print('Uncertainty in the temperature change of the water to its final temp.: {:f} °C'.format(err_delta_T_water))

Uncertainty in the mass of the water: 0.001414 g
Uncertainty in the mass of the ice: 0.001414 g
Uncertainty in the temperature change of ice to its melting point: 2.000000 °C
Uncertainty in the temperature change of melted ice to the final temp.: 0.300000 °C
Uncertainty in the temperature change of the water to its final temp.: 0.424264 °C


### Calculate the latent heat of fusion for water, *Hfus_water*. **(3 marks)**

In [14]:
# Calculate based on the equation you worked out
Hfus_water = (m_ice * cp_ice * T_ICE_C - cp_water * (m_ice * T_FINAL_C + m_water * delta_T_water)) / m_ice

# output the value
print('Heat of fusion for water: {:f} kJ kg⁻¹'.format(Hfus_water))

Heat of fusion for water: 329.736352 kJ kg⁻¹


### Calculate the uncertainty in your determination, *err_Hfus_water*. **(10 marks)**

In [15]:
# each component from equation 11 is worked out separately for clarity
denom_C = m_ice
num_C = np.zeros(7)
num_C[0] = err_m_ice * cp_water * m_water * delta_T_water / m_ice
num_C[1] = err_cp_ice * m_ice * T_ICE_C
num_C[2] = err_T_ICE_C * m_ice * cp_ice
num_C[3] = - err_cp_water * (m_ice * T_FINAL_C + m_water * delta_T_water)
num_C[4] = - err_m_water * cp_water * delta_T_water
num_C[5] = - err_T_FINAL_C * cp_water * (m_ice + m_water)
num_C[6] = err_T_WATER_C * cp_water * m_water
print(num_C)
err_Hfus_water = np.sqrt(np.sum(num_C ** 2)) / abs(denom_C)
print('Uncertainty in the heat of fusion of water: {:f} kJ kg⁻¹'.format(err_Hfus_water))

[ -0.61517265  -4.446204    74.82636      5.9908632    0.19999695
 -92.4680736   69.7815888 ]
Uncertainty in the heat of fusion of water: 7.641464 kJ kg⁻¹


### Replace the *Hfus_water* and *err_Hfus_water* to express the final latent heat value with its associated precision. Marks are awarded for correct rounding. **(2 marks)**

(*329.7* ± *7.6*) kJ kg⁻¹

### Explain how your value compares to the standard value for the heat of fusion of ice (334.744 kJ kg⁻¹), taking into account the uncertainty you determined for your measurement. **(1 mark)**

<div class="alert alert-block alert-info">
    <strong>Reasons:</strong>
    <ul>
        <li>The answer does <em>not</em> capture the standard value.</li>
        <li>There must be a systematic error.</li>
    </ul>
</div>

### Suggest one possible source of systematic error in a test like this that's different from your other answer. **(1 mark)**

<div class="alert alert-block alert-info">
    <br>
    Reasons for systematic error 
        <ul>
            <li>ice temperature not measured correctly</li>
            <li>impurities in the ice affect <i>T</i><sub><small>fus</small></sub> and ∆<sub><small>fus</small></sub><i>H</i></li>
             <li>The thermometer used in the experiment may be in contact with the styrofoam calorimeter. (Coffee cup Calorimeter with  a closed lid) </li>
        </ul>
        

###  Suggest a way by which the size of the error could be tested. **(2 marks)**

<div class="alert alert-block alert-info">
    <strong>Reasons</strong>
    <br>
        <ul>
            <li>ice temperature not measured correctly: use another thermometer</li>
            <li>salt in ice affect <i>T</i><sub><small>fus</small></sub> and ∆<sub><small>fus</small></sub>: chill ice another way, vary the amount of salt</li>
            <li>ice above <i>T</i><sub><small>fus</small></sub>: get new ice, try again</li>
            <li>all ice did not melt: need to use hotter water</li>
        </ul>
</div>